### PR TITLE
Add methods to get/set settings and some cleanup

### PR DESCRIPTION
**Pull request by xmakro**
Also applies rust-analyzer auto format

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -80,22 +114,20 @@ impl Ale {
...
 	/// Applies an action to the game and returns the reward.
-	/// 
+	///
 	/// It is the user's responsibility to check if the game has ended and reset
 	/// when necessary - this method will keep pressing buttons on the game over screen.
-	///
-	/// # Panics
-	/// If the action is not legal.
 	pub fn act(&mut self, action: i32) -> i32 {
-		assert!(self.legal_action_set().contains(&action), "Illegal action: {}", action);
```
Comment (trolleyman) [inline src/lib.rs:98]: Why was this removed? Is it needed for your use-case?

**Review by trolleyman — CHANGES_REQUESTED** [review]
Hey - thanks for the PR, looks all good to me, apart from one question I have about it.

e: Also not sure why the CI is failing.

File: src/lib.rs (hunk 1 above)
Comment (xmakro) [reply src/lib.rs:98]: Hey, sorry for the late reply, missed the notification for this. I'm also not sure about the failing CI, it seems to be unrelated to this PR.

I removed this to avoid copying the legal_action_set vector before every step and to match the C++ ALEInterface where illegal actions are converted into noops, see here:
https://github.com/mgbellemare/Arcade-Learning-Environment/blob/master/src/environment/stella_environment.cpp#L210

Comment (trolleyman) [reply src/lib.rs:98]: Sounds good to me, thanks :) I'll merge in and publish a new version
